### ISSUE TITLE
Prerendering failed because - MouseEvent is not defined

### ISSUE THREAD

**Issue by EvgenStudent** [MarkPieszak/aspnetcore-angular2-universal#417]
Hello!
I got the latest version of repo and when I run I get this error:
![image 2](https://user-images.githubusercontent.com/5446744/30660705-72546250-9e4a-11e7-87c1-f2323fced0b1.png)

**Comment by MarkPieszak** [comment]
Hey there! So it looks like you've added a library or code within your app that is using things that aren't available on the server. What were you trying to add in there?

Sometimes it can be as easy as only importing it in the browser NgModule and injecting it in components with @Optional(), other times you have to mock an implementation out for the server NgModule to use instead, and worst case you have to use webpack to manually swap out libraries and provide mocked classes/exports for them.

Let me know, so we can try and figure it out!

**Comment by casni14** [comment]
I've got the exact same problem with the latest version of the repo, haven't added anything just ran the code as described in the Readme

**Comment by MarkPieszak** [comment]
Oh wow really? Might be a third party library out of no where that updated in a strange way. I'll have to take a peak in the morning, very strange!

**Comment by casni14** [comment]
Indeed, thanks for the quick response 👍

**Comment by thomasm** [comment]
I've got the exact same problem. I did nothing else than downloading the project from the repository, and then tried to run it. I didn't dig too deep into the problem, but it seems like ngx-bootstrap causes the problem. I removed it from the project, and now it works. Except the ngx-Bootstrap demo page of course ;)

**Comment by casni14** [comment]
@thomasm how or what exactly did you remove? Just the declaration and routing part in app.module?

**Comment by thomasm** [comment]
@casni Yes, i removed the Ng2BootstrapModule and NgxBootstrapComponent imports and declarations, and the routing part in app.module.ts. I also removed the ngx-bootstrap folder in "app/containers/" , but that was probably not necessary.

**Comment by vyliniem** [comment]
Had the same problem. Culprit was ngx-bootstrap version v2.0.0-beta.4, fixing the dependency to version v2.0.0-beta.3 solved the issue.

**Comment by EvgenStudent** [comment]
About ngx-bootstrap package versions:
v2.0.0-beta.3 - doesn't work.
v2.0.0-beta.4 - works!
v2.0.0-beta.5 (latest) -  - doesn't work.

**Comment by davidsekar** [comment]
For me 1.9.3(Stable release) is working fine.

@MarkPieszak 
BTW, why we are trying to setup the beta? any reasons ?

**Comment by isaac2004** [comment]
@davidsekar there is a known bug where the non-beta version of bootstrap gives a template error in the console. vnext has  fix for this.

**Comment by frakevich** [comment]
Switching to another version of bootstrap takes care of the MouseEvents error, but gives another error:
NodeInvocationException: Prerendering failed because of error: TypeError: Cannot read property 'forRoot' of undefined

I tried with 1.9.3, and beta 03 & 04.

**Comment by MarkPieszak** [comment]
I'll pin it to beta.3 for now to fix the issue with anyone pulling a fresh copy of the repo, sorry about that guys! Third party libraries are never easy to deal with :)

@davidsekar I think someone set it up since it was complaining about 1.x versions since they used `<template>` tags instead of `<ng-template>` etc. I'll try to contact Dmitry the ngx-bootstrap author to let him know MouseEvent is breaking Universal.
